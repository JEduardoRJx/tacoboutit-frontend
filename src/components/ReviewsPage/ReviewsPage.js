import React, { Component } from 'react'
import { Text, View, Image, StyleSheet, Dimensions, Linking, TouchableOpacity, ScrollView, Modal, Button, KeyboardAvoidingView } from 'react-native';
import ReviewCard from '../ReviewCard/ReviewCard';
import AddReviewForm from '../AddReviewForm/AddReviewForm';
import { LinearGradient } from "expo-linear-gradient";


export default class ReviewsPages extends Component {
  state = {
    toggleForm: false
  }

  renderReviewCards = () => {
    const { tacos } = this.props;
    return tacos.map((taco) => {
      return <ReviewCard taco={taco} key={taco.id}/>;
    });
  }

  toggleReviewForm = () => {
    this.setState({toggleForm: !this.state.toggleForm})
  }

  toggleAddReviewButton = (tacos) => {
    if(tacos.length) {
      return (
        <TouchableOpacity onPress={() => this.toggleReviewForm ()} style={styles.buttonStyle}>
          {this.state.toggleForm ? <Text style={styles.buttonTextStyle}>Cancel</Text> : <Text style={styles.buttonTextStyle}>Add Review</Text>}
        </TouchableOpacity>
      )
    } else {
      return (<Text style={{color: 'white', fontSize: 32, textAlign: 'center'}}>Nothing To Review</Text>)
    }
  }
  
  render() {
    const { tacos, updateLocalReviews } = this.props;
    const reviewHeight = this.state.toggleForm ? 1 : 9
    return ( 
      <LinearGradient
        colors={["#F0CB35", "#D56C2C", "#C02425"]}
        style={styles.container}
      >
        <View style={{flex: reviewHeight}}>
          {/* {this.renderReviewCards()} */}
        </View>
        <KeyboardAvoidingView style={{flex: 1}}
          behavior="padding" 
          enabled
        >
          {(this.state.toggleForm && tacos.length > 0) && <AddReviewForm tacos={tacos} updateLocalReviews={updateLocalReviews} toggleReviewForm={this.toggleReviewForm}/>}
          {this.toggleAddReviewButton(tacos)}
          {/* <View style={{flex: 1.1}} /> */}
        </KeyboardAvoidingView>
      </LinearGradient>
    )
  }
}


const styles = StyleSheet.create({
  container: {height: Dimensions.get('window').height},
  buttonStyle: {flex: 1, backgroundColor: '#00BFFF', width: '100%', justifyContent: 'center', 
  alignItems: 'center'},
  buttonTextStyle: {color: 'white', fontSize: 32},
})