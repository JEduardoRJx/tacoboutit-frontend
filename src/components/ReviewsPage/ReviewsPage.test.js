import React from 'react';
import ReviewsPage from './ReviewsPage';
import { shallow } from 'enzyme';

describe('ReviewsPage', () => {
  const mockTacos = [
    {
        "id": 9,
        "type": "Testing testing",
        "restaurant_id": 2,
        "average_rating": 5.0,
        "reviews": [
            {
                "id": 1,
                "rating": 7.0,
                "review": "its ok",
                "taco": 9
            },
            {
                "id": 2,
                "rating": 4.0,
                "review": "lame",
                "taco": 9
            },
            {
                "id": 3,
                "rating": 4.0,
                "review": null,
                "taco": 9
            },
            {
                "id": 4,
                "rating": 1.0,
                "review": "This taco put my son in the hospital",
                "taco": 9
            },
            {
                "id": 5,
                "rating": 10.0,
                "review": "The best",
                "taco": 9
            },
            {
                "id": 6,
                "rating": 4.0,
                "review": "meh",
                "taco": 9
            }
        ]
    }
];
  const wrapper = shallow(<ReviewsPage tacos={mockTacos}/>);
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
    wrapper.setState({ toggleForm: true });
    expect(wrapper).toMatchSnapshot();
    wrapper.setState({ toggleForm: false });
  });
  it('toggleReviewForm should toggle toggleForm property in state', () => {
    expect(wrapper.state('toggleForm')).toEqual(false);
    wrapper.instance().toggleReviewForm();
    expect(wrapper.state('toggleForm')).toEqual(true);
  });
});