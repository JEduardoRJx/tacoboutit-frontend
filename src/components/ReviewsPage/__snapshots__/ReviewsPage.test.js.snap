// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewsPage should match the snapshot 1`] = `
<LinearGradient
  colors={
    Array [
      "#F0CB35",
      "#D56C2C",
      "#C02425",
    ]
  }
  style={
    Object {
      "height": 1334,
    }
  }
>
  <View
    style={
      Object {
        "flex": 9,
      }
    }
  >
    <ScrollViewMock>
      <ReviewCard
        key="9"
        taco={
          Object {
            "average_rating": 5,
            "id": 9,
            "restaurant_id": 2,
            "reviews": Array [
              Object {
                "id": 1,
                "rating": 7,
                "review": "its ok",
                "taco": 9,
              },
              Object {
                "id": 2,
                "rating": 4,
                "review": "lame",
                "taco": 9,
              },
              Object {
                "id": 3,
                "rating": 4,
                "review": null,
                "taco": 9,
              },
              Object {
                "id": 4,
                "rating": 1,
                "review": "This taco put my son in the hospital",
                "taco": 9,
              },
              Object {
                "id": 5,
                "rating": 10,
                "review": "The best",
                "taco": 9,
              },
              Object {
                "id": 6,
                "rating": 4,
                "review": "meh",
                "taco": 9,
              },
            ],
            "type": "Testing testing",
          }
        }
      />
    </ScrollViewMock>
  </View>
  <KeyboardAvoidingView
    behavior="padding"
    enabled={true}
    keyboardVerticalOffset={0}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#00BFFF",
          "flex": 1,
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 32,
          }
        }
      >
        Add Review
      </Text>
    </TouchableOpacity>
  </KeyboardAvoidingView>
</LinearGradient>
`;

exports[`ReviewsPage should match the snapshot 2`] = `
<LinearGradient
  colors={
    Array [
      "#F0CB35",
      "#D56C2C",
      "#C02425",
    ]
  }
  style={
    Object {
      "height": 1334,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <ScrollViewMock>
      <ReviewCard
        key="9"
        taco={
          Object {
            "average_rating": 5,
            "id": 9,
            "restaurant_id": 2,
            "reviews": Array [
              Object {
                "id": 1,
                "rating": 7,
                "review": "its ok",
                "taco": 9,
              },
              Object {
                "id": 2,
                "rating": 4,
                "review": "lame",
                "taco": 9,
              },
              Object {
                "id": 3,
                "rating": 4,
                "review": null,
                "taco": 9,
              },
              Object {
                "id": 4,
                "rating": 1,
                "review": "This taco put my son in the hospital",
                "taco": 9,
              },
              Object {
                "id": 5,
                "rating": 10,
                "review": "The best",
                "taco": 9,
              },
              Object {
                "id": 6,
                "rating": 4,
                "review": "meh",
                "taco": 9,
              },
            ],
            "type": "Testing testing",
          }
        }
      />
    </ScrollViewMock>
  </View>
  <KeyboardAvoidingView
    behavior="padding"
    enabled={true}
    keyboardVerticalOffset={0}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <AddReviewForm
      tacos={
        Array [
          Object {
            "average_rating": 5,
            "id": 9,
            "restaurant_id": 2,
            "reviews": Array [
              Object {
                "id": 1,
                "rating": 7,
                "review": "its ok",
                "taco": 9,
              },
              Object {
                "id": 2,
                "rating": 4,
                "review": "lame",
                "taco": 9,
              },
              Object {
                "id": 3,
                "rating": 4,
                "review": null,
                "taco": 9,
              },
              Object {
                "id": 4,
                "rating": 1,
                "review": "This taco put my son in the hospital",
                "taco": 9,
              },
              Object {
                "id": 5,
                "rating": 10,
                "review": "The best",
                "taco": 9,
              },
              Object {
                "id": 6,
                "rating": 4,
                "review": "meh",
                "taco": 9,
              },
            ],
            "type": "Testing testing",
          },
        ]
      }
      toggleReviewForm={[Function]}
    />
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#00BFFF",
          "flex": 1,
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 32,
          }
        }
      >
        Cancel
      </Text>
    </TouchableOpacity>
  </KeyboardAvoidingView>
</LinearGradient>
`;
